File: solutions/beecrowd/1983/1983.cpp
#include <cstdint>
#include <iostream>

int main() {
    std::int16_t n;
    std::int32_t m, bigest_m;
    float grade, bigest_grade;

    std::cin >> n;
    std::cin >> bigest_m >> bigest_grade;
    n--;

    while (n--) {
        std::cin >> m >> grade;

        if (grade > bigest_grade) {
            bigest_grade = grade;
            bigest_m = m;
        }
    }

    if (bigest_grade < 8.0) {
        std::cout << "Minimum note not reached" << std::endl;
    } else {
        std::cout << bigest_m << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1983/1983.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

n = int(input())
biggest_m, biggest_grade = input().split()
biggest_m, biggest_grade = int(biggest_m), float(biggest_grade)

for _ in range(n - 1):
    m, grade = input().split()
    m, grade = int(m), float(grade)

    if grade > biggest_grade:
        biggest_m = m
        biggest_grade = grade

if biggest_grade < 8.0:
    output('Minimum note not reached\n')
else:
    output(f'{biggest_m}\n')
